### PR TITLE
Adding implementation for separate ui database user

### LINKED ISSUE #224: Allow Engine/UI Database User separation
The UI database connection is re-using the database information provided by the engine. Which is expected given it likely requires the same data/permissions.

As an initial security measure, being able to bootstrap and utilize a separate database user for engine and UI would allow tracing access to the database more finitely given a breach.  

Follow-on issues may help provide insight on levels of permissions required and the possibility of separating permissions within the database.

### PR DESCRIPTION
**Pull request by brandtkeller**
Closes #224 

Relates to #195 and #196 

Directly implements a configuration to allow overriding the enterpriseUI database user and providing separation for access to the database by Engine/UI to aid in traceability/logging/auditing.

Indirectly addresses some standardization of DB host across resource files. 

Willing to consider another templating mechanism outside of the ternary - it's self-documenting but not syntactically pretty.

### CODE REVIEW

**Comment by Btodhunter** [comment]
Thanks for this PR @brandtkeller, it seems super useful! I need to do some manual testing on this change before merging it, but probably won't be able to get to that testing until sometime next week.

**Comment by flickerfly** [comment]
@Btodhunter Is there any support that can be given to help get this over the hump? We're currently manually managing the secret to get this same result. Having a UI specific db user is working fine, but we haven't been able to trim down the level of permissions that may be required for that specific user yet.

File: stable/anchore-engine/templates/enterprise_ui_secret.yaml
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_ui_secret.yaml:18]: @brandtkeller if `.Values.anchoreEnterpriseUi.dbUser` is set, should we be handling the password separately as well? Using `.Values.postgresql.postgresPassword`, which is shared with the engine db user, doesn't seem ideal.

**Comment by Btodhunter** [comment]
@brandtkeller I'll keep my eye on this and get it through asap.

File: stable/anchore-engine/templates/enterprise_ui_secret.yaml
Comment (brandtkeller) [reply stable/anchore-engine/templates/enterprise_ui_secret.yaml:18]: I believe that is a fair assessment. We're trying to establish some ability to separate the users and their credentials - it would make sense to allow a non-shared password.

Comment (brandtkeller) [reply stable/anchore-engine/templates/enterprise_ui_secret.yaml]: Added support for both user/password to be unique. It's a lengthy template with multiple ternaries - so i'm open to input or anyone doing direct modification.

**Comment by Btodhunter** [comment]
@brandtkeller thanks for the update! I'm reviewing the template now. Can you bump the chart version? Then I'll merge this after finishing the review.

**Comment by Btodhunter** [comment]
@brandtkeller you may also need to rebase on `main` to get the latest set of required tests in your github config.

**Review by Btodhunter — APPROVED** [review]
lgtm!

**Pull request merged**
